python/cudf/dataframe/dataframe.py

```diff
@@ -970,8 +993,43 @@ def _sanitize_values(self, series, SCALAR):
         """
         if SCALAR:
             col = series
-        if not isinstance(series, Series):
+
+        from collections.abc import Sequence
+
+        def is_list_like(obj):
+            if isinstance(obj, (Sequence,)) and \
+                    not isinstance(obj, (str, bytes, int)):
+                return True
+            else:
+                return False
+
+        if not isinstance(series, Series) and is_list_like(series):
+            '''
+            Case when series is not a series and list-like.
+            Reason we will have to guard this with not Series check
+            is because we are converting non-scalars to cudf Series.
+            '''
             series = Series(series)
+        elif is_list_like(series):
+            # Case when series is scalar and of type list.
+            series = Series(series)
+        elif isinstance(series, Series):
+            # Case when series is a cudf Series
+            series = series
+        else:
+            # Case when series is just a non-list
+            series = Series(series)
+            if (len(self._index) == 0) and (series.index > 0) \
+                    and len(self.columns) == 0:
+                # When self has 0 columns and series has values
+                # we can safely go ahead and assign.
+                return series
+            elif (len(self._index) == 0) and (series.index > 0) \
```

Comment:
Use `self.index` instead of `self._index`

Reply:
Changed to `.index`